llmfoundry/data/finetuning/tasks.py

```diff
@@ -340,18 +348,21 @@ def dataset_mapper(example: Dict):
             return _tokenize_formatted_example(example, tokenizer)
 
         detected_cpu_count = os.cpu_count() or 1
-        num_cpus_to_use = max(1, detected_cpu_count - 4)
+        detected_cpus_with_margin = detected_cpu_count - 8
```

Comment:
Can you add a comment in the code on why we are doing this? It's a good selection since we don't want to overload all the CPUs I suppose.